Repository/AssetStorageRepository.php

```diff
@@ -54,14 +54,24 @@ public function clearCache(): bool
         }
     }
 
-    public function removeByHash(string $hash): bool
+    public function removeByHash(string $hash, ?string $context = null): bool
     {
         try {
             $qb = $this->createQueryBuilder('asset')->delete();
             $qb->where($qb->expr()->eq('asset.hash', ':hash'));
-            $qb->setParameters([
-                ':hash' => $hash,
-            ]);
+
+            if ($context !== null) {
+                $qb->andWhere($qb->expr()->eq('asset.context', ':context'));
+                $qb->setParameters([
+                    ':hash' => $hash,
+                    ':context' => $context,
+                ]);
+            } else {
```

Comment:
To prevent complexity (counted by the number of if/else statements used) and reduce code duplication (:hash => hash) I would suggest using the $qb->setParameter('key', value) method.

```
$qb->setParameter(:hash',$hash);
if ($context !== null) {
    $qb->andWhere($qb->expr()->eq('asset.context', ':context'));
    $qb->setParameter(':context', $context);
}
```

Another benefit: only half the number of lines of code needed!

Reply:
If fixed if as recommended.